### PR TITLE
CDAP-12437 call destroy on alert publishers

### PR DESCRIPTION
**Pull request by albertshau**

### CODE REVIEW

File: cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/co/cask/cdap/datapipeline/SmartWorkflow.java [hunk 1]
```diff
@@ -334,6 +334,12 @@ public void destroy() {
         alertPublisher.publish(trackedIterator);
       } catch (Exception e) {
         LOG.warn("Stage {} had errors publishing alerts. Alerts may not have been published.", name, e);
+      } finally {
+        try {
+          alertPublisher.destroy();
```
Comment (anwar6953) [inline cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/co/cask/cdap/datapipeline/SmartWorkflow.java:339]: we call it already in spark. we should remove that then, right? Or do we not use SmartWorkflow in that case?

Comment (albertshau) [reply cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/co/cask/cdap/datapipeline/SmartWorkflow.java:339]: this section doesn't get run unless it's mapreduce

**Comment by albertshau** [comment]
https://builds.cask.co/browse/CDAP-RUT1289

File: cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/co/cask/cdap/datapipeline/SmartWorkflow.java (hunk 1 above)
Comment (anwar6953) [reply cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/co/cask/cdap/datapipeline/SmartWorkflow.java:339]: Ah ok. That wasn't clear to me.

**Review by anwar6953 — APPROVED** [review]

**Comment by sreevatsanraman** [comment]
Build passed. Merging.

**Pull request merged**